src/cls/rbd/cls_rbd_types.h

```diff
@@ -733,6 +939,153 @@ struct SnapshotInfo {
 };
 WRITE_CLASS_ENCODER(SnapshotInfo);
 
+enum GroupSnapshotNamespaceType {
+  GROUP_SNAPSHOT_NAMESPACE_TYPE_USER   = 0,
+  GROUP_SNAPSHOT_NAMESPACE_TYPE_MIRROR = 1,
+};
+
+struct UserGroupSnapshotNamespace {
```

Comment:
Nit:  Rename  the group snapshot namespaces to
GroupUserSnapshotNamespace
GroupMirrorSnapshotNamespace
GroupUnknownSnapshotNamespace
?

Reply:
Starting the type name with Group makes it easier to understand that they refer to SnapshotNamespaces for group snaps.

Reply:
> ... so add Image or Group at the front per Nithya, but also move User, Group, Trash or Mirror to the back.

carrying this statement from above. This perfectly aligns with what we have now correct?

<IMAGE/GROUP>_<SNAPSHOT_NAMESPACE_TYPE>_<USER, GROUP, TRASH OR MIRROR>

Reply:
I've renamed the GroupSnapshotNamespace types and  `ImageGroupSnapshotNamespace` to `ImageSnapshotNamespaceGroup`. I'm evaluating how big the changes to rename all the ImageSnapshotNamespaces will be. It might be better to do that as a separate PR.

Reply:
> > ... so add Image or Group at the front per Nithya, but also move User, Group, Trash or Mirror to the back.
> 
> carrying this statement from above. This perfectly aligns with what we have now correct?
> 
> <IMAGE/GROUP>_<SNAPSHOT_NAMESPACE_TYPE>_<USER, GROUP, TRASH OR MIRROR>

Not sure I understand this - this is not the current format used in the code.